### PR TITLE
Rebased version of #125

### PR DESCRIPTION
**Pull request by jugglerchris**

### CODE REVIEW

File: src/render/text_renderer.rs [hunk 1]
```diff
@@ -169,6 +169,18 @@ impl<T: Debug + Eq + PartialEq + Clone + Default> TaggedLine<T> {
         true
     }
 
+    /// Return true if the line is non-empty
```
Comment (jugglerchris) [inline src/render/text_renderer.rs:172]: I think the comments is backwards.

Comment (jugglerchris) [reply src/render/text_renderer.rs]: Fixed.

**Comment by jugglerchris** [comment]
@sftse This looks good to me.  Do you have any concerns about how it works now?

> I don't quite get how this works with deeply nested tables (see deeply_nested test), but think this could benefit from your input at this stage.

Do you mean the "vertical row" feature?  (If columns don't fit next to each other, fall back to putting them on top of each other with "////" as a border to try to show they belong side by side)

That's a good point, thinking about it.  Maybe raw mode doesn't need much more than:

```diffdiff --git a/src/lib.rs b/src/lib.rs
index 6959faf..0bd346d 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1834,7 +1834,7 @@ fn render_table_tree<T: Write, D: TextDecorator>(
         + col_sizes.len().saturating_sub(1);
     let width = renderer.width();
 
-    let vert_row = min_size > width || width == 0;
+    let vert_row = renderer.options.raw || (min_size > width || width == 0);
 
     let mut col_widths: Vec<usize> = if !vert_row {
         col_sizes
```

along with `!draw_borders`.

**Comment by jugglerchris** [comment]
Ok, I've updated to use vertical rows instead of a separate loop over rows.
I've also removed some of the `if raw` tests when making subrenderers - in most cases they come for free because the `prefix`es are zero length from `TrivialDecorator`.  There was  one case (Dd) with a hard-coded prefix though.
Finally, the output from the test is slightly different - there's no trailing whitespace (which was there because normal table columns are padded, but that's not needed for the "vertical" columns).  I think that's probably better.

**Comment by sftse** [comment]
I can't push to this branch, but this idea turned out great. I reverted even more of my previous changes, see #129
